test/run-test.ps1

```diff
@@ -0,0 +1,65 @@
+[cmdletbinding()]
+param(
+   [string]$OS="windowsservercore"
+)
+
+Set-StrictMode -Version Latest
+$ErrorActionPreference="Stop"
+
+$dockerRepo="microsoft/dotnet-nightly"
+$repoRoot = Split-Path -parent $PSScriptRoot
+
+if ($env:DEBUGTEST -eq $null) {
+    $optionalDockerRunArgs="--rm"
+}
+else {
+    $optionalDockerRunArgs=""
+}
+
+pushd $repoRoot
+
+Get-ChildItem -Directory -Exclude 'test', 'update-dependencies', '.*' | foreach {
+    $tagBase="$($dockerRepo):$($_.Name)-$OS"
+
+    $timeStamp = Get-Date -Format o | foreach {$_ -replace ":", "." -replace "+", "-"}
+    $appName="app$timeStamp"
+    $appDir="${repoRoot}\.test-assets\${appName}"
+
+    New-Item $appDir -type directory | Out-Null
+
+    Write-Host "----- Testing $tagBase-sdk -----$repoRoot   $appDir"
+    docker run -t $optionalDockerRunArgs -v "$($appDir):c:\$appName" -v "$repoRoot\test:c:\test" --name "sdk-test-$appName" --entrypoint powershell "$tagBase-sdk" c:\test\create-run-publish-app.ps1 "c:\$appName"
+    if (-NOT $?) {
+        throw  "Testing $tagBase-sdk failed"
+    }
+
+    Write-Host "----- Testing $tagBase-core -----"
+    docker run -t $optionalDockerRunArgs -v "$($appDir):c:\$appName" --name "core-test-$appName" --entrypoint dotnet "$tagBase-core" "C:\$appName\publish\$appName.dll"
+    if (-NOT $?) {
+        throw  "Testing $tagBase-core failed"
+    }
+
+    Write-Host "----- Testing $tagBase-onbuild -----"
+    pushd $appDir
+    $onbuildTag = "$tagBase-onbuild".ToLowerInvariant()
```

Comment:
The bash version is incorrect IMO.  The purpose of the statement is to convey at a higher level that the xxx-onbuild image is being tested rather than literally stating the image that is being used to test with.  I will update the Bash version.